04-store-api/starter/controllers/products.js

```diff
@@ -0,0 +1,65 @@
+const Product = require("../models/product");
+
+const getAllProducts = async (req, res) => {
+   const { featured, company, name, sort, fields, numericFilters } = req.query;
+   const queryObj = {};
+
+   if (featured) {
+      queryObj.featured = featured === "true" ? true : false;
```

Comment:
This is not wrong, but you really wouldn't see this. You can short circuit this by just checking if it is truthy:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND

Give it a try and let me know if you have a question about it

Reply:
You are absolutely right, from now on I will not only follow the videotutorial but I will try to review it in detail to optimize it. Thank you very much Hector.